pkg/ibmvpc/vpc.go

```diff
@@ -866,40 +867,94 @@ func actionNameStr(action *string) (string, error) {
+			if !dst.IPBlock().Intersect(ipBlock).IsEmpty() {
+				if _, ok := trCnToRulesInTable[trCnPrefixes.Table]; !ok {
+					trCnToRulesInTable[trCnPrefixes.Table] = []vpcmodel.RulesInTable{}
+				}
+				trCnToRulesInTable[trCnPrefixes.Table] = append(trCnToRulesInTable[trCnPrefixes.Table],
+					trCnPrefixes)
 			}
 		}
 	}
-	return nil
+	aggregatedRes := []vpcmodel.RulesInTable{}
+	// 2. Aggregates RulesInTable per transit connection
+	// trCnPrefixes is from "atomic" IPBlocks, each has exactly one prefix filter of transit connection
+	// that determines its relevant routing.
+	// However, once https://github.com/np-guard/vpc-network-config-analyzer/issues/576 is solved then
+	// there could be few "atomic" IPBlocks relevant for <src, dst>, possibly one allow and one deny
+	for trCn, trCnRulesInTable := range trCnToRulesInTable {
+		prefixesTrCn := vpcmodel.RulesInTable{Table: trCn, Rules: []int{}, RulesOfType: vpcmodel.NoRules}
+		for _, prefixEntry := range trCnRulesInTable {
+			prefixesTrCn.Rules = append(prefixesTrCn.Rules, prefixEntry.Rules...)
+			if prefixesTrCn.RulesOfType == vpcmodel.NoRules { // first time
+				prefixesTrCn.RulesOfType = prefixEntry.RulesOfType
+			} else if prefixesTrCn.RulesOfType != vpcmodel.BothAllowDeny &&
+				prefixesTrCn.RulesOfType != prefixEntry.RulesOfType {
+				// one is OnlyDeny and the other OnlyAllow
+				prefixesTrCn.RulesOfType = vpcmodel.BothAllowDeny
+			}
+			aggregatedRes = append(aggregatedRes, prefixesTrCn)
+		}
+	}
+	return aggregatedRes
+}
+
+func (tgw *TransitGateway) StringDetailsOfRules(listRulesInTransitConns []vpcmodel.RulesInTable, verbose bool) (string, error) {
+	strRes := []string{}
+	for _, prefixesInTransitConn := range listRulesInTransitConns {
+		transitConn := tgw.tgwConnList[prefixesInTransitConn.Table]
+		if verbose {
+			for _, prefixInTransConnIndx := range prefixesInTransitConn.Rules {
+				thisPrefixStr := ""
+				tgwRouterFilterDetails, actionName, err := tgw.tgwPrefixStr(transitConn, prefixInTransConnIndx)
+				if err != nil {
+					return "", err
+				}
+				action := ""
+				if actionName == permitAction {
+					action = "allows"
+				} else {
+					action = "blocks"
+				}
+				thisPrefixStr = fmt.Sprintf("transit gateway %s %s connection with the following prefix\n\t%s\n",
+					tgw.Name(), action, tgwRouterFilterDetails)
+				strRes = append(strRes, thisPrefixStr)
+			}
+		} else {
+			{
+				thisConnectionStr := ""
+				noVerboseStr := fmt.Sprintf("cross-vpc-connection: transit-connection %s of transit-gateway %s ", *transitConn.Name, tgw.Name())
+				switch prefixesInTransitConn.RulesOfType {
+				case vpcmodel.OnlyAllow:
+					thisConnectionStr = noVerboseStr + "allows connection"
+				case vpcmodel.OnlyDeny:
+					thisConnectionStr = noVerboseStr + "denies connection"
+				case vpcmodel.BothAllowDeny:
+					thisConnectionStr = noVerboseStr + "partly allows connection"
+				}
+				strRes = append(strRes, thisConnectionStr)
```

Comment:
an example is required to demonstrate the output in such case ("partly allows" and its verbose version) , so that can better understand and review the output generated here..

Reply:
"partly allow" will be possible only after #576 is resolved; this is the discussion we had in the slack channel last week. 
We can prioritize this issue; another option is to not support this option until the issue is solved.

Reply:
I've solved #576 and will add an example and test

Reply:
It turns out that BothAllowDeny is not possible; this is since for tgw based routing <src, dst> are an endpoint each, with a single IP. I removed redundant code